### DISCUSSION TITLE
Installing additional python modules

### DISCUSSION THREAD

**Discussion by jjsvs** [open-webui/open-webui#2435] [Q&A] [answered]
The code execution is really nice feature to have, but I was wondering can additional python modules be installed like pygame or any other module which the generated code uses. I tried installing pygame in the docker container and it was installed successfully but when I tried to run the generated code in open web Ui, It still gave me the module not found error, is there something I am doing wrong or is there any other way to add the module

**Comment by justinh-rahb** [comment, marked as answer]
Unfortunately with the way this is currently implemented via `Pyodide` we're at the mercy of the modules that can be supported in that.

https://pyodide.org/en/stable/usage/packages-in-pyodide.html

**Reply by jjsvs** [reply]
well I was under the impression that it was implemented in a sandbox or just use the container as a base, anyway thanks for the reply.

**Reply by justinh-rahb** [reply]
Pyodide runs as a WASM executable in your browser, 100% local so it is sandboxed in the sense that any issues that arise will be solely isolated to the client 😆
